pkg/controller/runGC.go

```diff
@@ -0,0 +1,143 @@
+package controller
+
+import (
+	"context"
+	"sort"
+
+	hephv1 "github.com/dominodatalab/hephaestus/pkg/api/hephaestus/v1"
+	"github.com/dominodatalab/hephaestus/pkg/clientset"
+	"github.com/dominodatalab/hephaestus/pkg/config"
+	"github.com/dominodatalab/hephaestus/pkg/kubernetes"
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrlzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type ImageBuildGC struct {
+	maxIBRetention int
+	hephClient     clientset.Interface
+	namespaces     []string
+}
+
+func NewImageBuildGC(maxIBRetention int, log logr.Logger, ibNamespaces []string) (
+	*ImageBuildGC, error) {
+	log.Info("Initializing Kubernetes Hephaestus V1 client")
+
+	config, err := kubernetes.RestConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	hephClient, _ := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	var ns []string
+	if len(ibNamespaces) > 0 {
+		log.Info("Limiting GC cleanup", "namespaces", ibNamespaces)
+		ns = ibNamespaces
+	} else {
+		ns, err = getAllNamespaces(config)
+		if err != nil {
+			log.Info("Unable to access cluster namespaces")
+			return nil, err
+		}
+		log.Info("Watching all namespaces")
+	}
+
+	return &ImageBuildGC{
+		maxIBRetention: maxIBRetention,
+		hephClient:     hephClient,
+		namespaces:     ns,
+	}, nil
+}
+
+func getAllNamespaces(config *rest.Config) ([]string, error) {
+	// create k8s client to access all namespaces in the cluster
+	k8sClient, err := k8s.NewForConfig(config)
```

Comment:
I was also thinking another route would be to add the k8s client to the Hephaestus client. Seemed weird to change the client for this thought